uxarray/grid/area.py

```diff
@@ -56,23 +65,30 @@ def calculate_face_area(
     # num triangles is two less than the total number of nodes
     num_triangles = num_nodes - 2
 
+    if coords_type == "spherical":
```

Comment:
Probably refer to how @philipc2 did about the coordinate conversion will be better.

The area correction function will only take cartesian xyz as input. And a wrapper/pretreatment function can be used before calling the area correction to handle the coordinate conversion if needed.

Uxarray is using all cartesian coordinates for internal algorithms anyway. The spherical coordinates is mostly for users

Reply:
Can be done, but this is general area correction stuff and can be addressed later, not a subject of this PR.

Reply:
@philipc2 Do you think the current implementation is fine for this release? I am worrying that once we call this function in zonal average and calculate huge amounts of faces, it might cause performance issues here and we need to redesign the codes again and again.